### PR TITLE
ksh: fix cross build

### PR DESCRIPTION
**Pull request by Gottox**

### CODE REVIEW

File: srcpkgs/ksh/patches/fix-cross.diff [hunk 1]
```diff
@@ -0,0 +1,9 @@
+--- scripts/libast_prereq.sh.orig	2021-03-31 22:45:26.222481811 +0200
++++ scripts/libast_prereq.sh	2021-03-31 22:45:34.162481713 +0200
+@@ -25,5 +25,5 @@ cd "$MESON_BUILD_ROOT"
+ # Generate header files whose content depends on the current platform.
+ "$MESON_SOURCE_ROOT/scripts/siglist.sh" > features/siglist.h
+ # shellcheck disable=SC2086
+-$CC -D_BLD_DLL $INC_DIRS -std=gnu99 -o "sfinit" "$MESON_SOURCE_ROOT/etc/sfinit.c"
++cc -D_BLD_DLL $INC_DIRS -std=gnu99 -o "sfinit" "$MESON_SOURCE_ROOT/etc/sfinit.c"
```
Comment (ericonr) [inline srcpkgs/ksh/patches/fix-cross.diff:8]: Is this safe?

>  Generate header files whose content depends on the current platform.

Worries me slightly :/

File: srcpkgs/ksh/template [hunk 2]
```diff
@@ -4,14 +4,15 @@ version=2020.0.0
 revision=1
 wrksrc="ast-${version}"
 build_style=meson
-configure_args="-Dbuild-api-tests=false -Dbuild-api-tests-only=true"
+build_helper="qemu"
+configure_args="-Dbuild-api-tests=false -Dbuild-api-tests-only=true
+ -Dexe_wrapper=qemu-${XBPS_TARGET_QEMU_MACHINE}-static"
```
Comment (ericonr) [inline srcpkgs/ksh/template:9]: I think this is added by default by the build style, isn't it? In the toolchain file.

File: srcpkgs/ksh/patches/fix-cross.diff (hunk 1 above)
Comment (Gottox) [reply srcpkgs/ksh/patches/fix-cross.diff:8]: You're right. I thought the build just relied on ast_config.h, but there are other compiler macros used.

**Comment by Gottox** [comment]
Will be fixed with the new release.

**Pull request closed without merging**